### PR TITLE
Optimize and open parse_hex_digit in strutils.asm

### PR DESCRIPTION
**Pull request by sijnstra**
Multiple routines optimized and hex digit parsing exposed for other routines to use.

### CODE REVIEW

File: romdisk/strutils.asm [hunk 1]
```diff
@@ -418,15 +432,11 @@ byte_to_ascii:
...
-        sub 10
-        jp nc, _byte_to_ascii_af
-        ; Byte is between 0 and 9
-        add '0' + 10
-        ret
-_byte_to_ascii_af:
-        ; Byte is between A and F
-        add 'A'
+        ; efficient routine to convert nibble into ASCII
+        add a,090h
```
Comment (Zeal8bit) [inline romdisk/strutils.asm:436]:
```suggestion
        add a, 0x90
```
Just to be consistent with all the other files, can we use C syntax?

**Review by Zeal8bit — COMMENTED** [review]
Thank you very much for these improvements!

I left a nitpick but looks very good!

**Comment by sijnstra** [comment]
Reasonable nit-pick to be honest - always nice to have code remain consistent for people to be able to read later.

**Comment by Zeal8bit** [comment]
Thanks for the modifications, merging!

**Pull request merged**